infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/plugins/DefaultAdapterForUpgrade.java

```diff
@@ -12,33 +16,155 @@ public abstract class DefaultAdapterForUpgrade implements ILogReplicationVersion
 
     public static final String NAMESPACE = "LR-Test";
 
+    public static final String PINNED_VERSION_KEY = "PINNED_VERSION_KEY";
+
+    public static final String LATEST_VERSION = "LATEST-VERSION";
+
+    /* Represents the current code's version
+     */
+    public static final String NODE_VERSION_KEY = "VERSION";
+
     String versionString = "version_latest";
+
+    String pinnedVersionString = "version_former";
     CorfuStore corfuStore;
 
-    @Override
-    public String getVersion() {
-        if (corfuStore != null) {
-            try {
-                corfuStore.openTable(NAMESPACE, VERSION_TEST_TABLE,
-                        LogReplicationStreams.VersionString.class,
-                        LogReplicationStreams.Version.class, CommonTypes.Uuid.class,
-                        TableOptions.builder().build());
-            } catch (Exception e) {
-                // Just for wrap this up
-            }
+    public DefaultAdapterForUpgrade(CorfuRuntime runtime) {
+        this.corfuStore = new CorfuStore(runtime);
+        try {
```

Comment:
nit: can we reuse openVersionTable() here?

Reply:
Thanks, done.